python/GafferSceneUI/HierarchyView.py

```diff
@@ -356,3 +359,121 @@ def _updateFromPathFilter( self ) :
 	def __patternEditingFinished( self, widget ) :
 
 		self.pathFilter().setMatchPattern( self.__patternWidget.getText() )
+
+##########################################################################
+# _ExpansionWidget
+##########################################################################
+
+class _ExpansionWidget( GafferUI.Widget ) :
+
+	def __init__( self ) :
+
+		button = GafferUI.MenuButton(
+			image = "locationExpanded.png",
+			hasFrame = False,
+			toolTip = "Expansion",
+			menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ), title = "Expansion" )
+		)
+
+		GafferUI.Widget.__init__( self, button )
+
+	def __menuDefinition( self ) :
+
+		menu = IECore.MenuDefinition()
+
+		menu.append(
+			"/Bookmarks",
+			{
+				"subMenu" : Gaffer.WeakMethod( self.__bookmarkMenuDefinition )
+			}
+		)
+
+		return menu
+
+	def __bookmarkMenuDefinition( self ) :
+
+		scriptNode = self.ancestor( GafferUI.ScriptWindow ).scriptNode()
```

Comment:
This works, but I'd suggest `ancestor( GafferUI.Editor ).scriptNode()` as a closer source of the same info.

Reply:
Updated in 13796ed.